From 3d7fa1dc1c9ea7e03abc89d7b1b2193305564fab Mon Sep 17 00:00:00 2001
From: Terry Lv <r65388@freescale.com>
Date: Fri, 12 Aug 2011 12:49:30 +0800
Subject: [PATCH 340/691] ENGR00154666-1: Align u-boot mmc command with
 community

Modifications on config files.
1. Remove CONFIG_BOOT_PARTITION_ACCESS.
2. Change all mmc read command in default env configs.

Signed-off-by: Terry Lv <r65388@freescale.com>
---
 include/configs/mx28_evk.h               |    5 +++--
 include/configs/mx50_arm2.h              |    5 +++--
 include/configs/mx50_arm2_ddr2.h         |    1 -
 include/configs/mx50_arm2_lpddr2.h       |    5 +++--
 include/configs/mx50_arm2_mfg.h          |    9 ++++-----
 include/configs/mx50_rd3.h               |    5 +++--
 include/configs/mx50_rd3_android.h       |    9 +++++----
 include/configs/mx50_rd3_mfg.h           |    7 +++----
 include/configs/mx50_rdp.h               |    5 +++--
 include/configs/mx50_rdp_android.h       |    4 ++--
 include/configs/mx50_rdp_mfg.h           |    8 ++++----
 include/configs/mx51_3stack_android.h    |    7 ++++---
 include/configs/mx51_bbg_android.h       |    8 +++++---
 include/configs/mx53_ard.h               |    5 ++---
 include/configs/mx53_ard_ddr3.h          |    5 ++---
 include/configs/mx53_arm2_ddr3.h         |    5 +++--
 include/configs/mx53_arm2_ddr3_android.h |   23 ++++++++++++-----------
 include/configs/mx53_evk.h               |    6 ++++--
 include/configs/mx53_evk_android.h       |   24 ++++++++++++------------
 include/configs/mx53_evk_mfg.h           |    1 -
 include/configs/mx53_loco.h              |    4 ++--
 include/configs/mx53_loco_mfg.h          |    1 -
 include/configs/mx53_pcba_android.h      |    6 +++---
 include/configs/mx53_smd.h               |    3 ++-
 include/configs/mx53_smd_android.h       |    9 +++++----
 include/configs/mx53_smd_mfg.h           |    1 -
 include/configs/mx6q_sabreauto.h         |    5 +++--
 27 files changed, 92 insertions(+), 84 deletions(-)

diff --git a/include/configs/mx28_evk.h b/include/configs/mx28_evk.h
index 557c24d..feab354 100644
--- a/include/configs/mx28_evk.h
+++ b/include/configs/mx28_evk.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2010 Freescale Semiconductor, Inc.
+ * Copyright (C) 2010-2011 Freescale Semiconductor, Inc.
  *
  * This program is free software; you can redistribute it and/or
  * modify it under the terms of the GNU General Public License as
@@ -84,7 +84,8 @@
 	"bootargs_mmc=setenv bootargs ${bootargs} root=/dev/mmcblk0p3 " \
 		"rw rootwait ip=dhcp fec_mac=${ethaddr}\0" \
 	"bootcmd_mmc=run bootargs_mmc; " \
-		"mmc read 0 ${loadaddr} 100 3000; bootm\0" \
+		"mmc dev 0"	\
+		"mmc read ${loadaddr} 100 3000; bootm\0" \
 
 /*
  * U-Boot Commands
diff --git a/include/configs/mx50_arm2.h b/include/configs/mx50_arm2.h
index c4895db..e90fcb5 100644
--- a/include/configs/mx50_arm2.h
+++ b/include/configs/mx50_arm2.h
@@ -123,7 +123,9 @@
 			"tftpboot ${loadaddr} ${kernel}; bootm\0"	\
 		"bootargs_mmc=setenv bootargs ${bootargs} ip=dhcp "     \
 			"root=/dev/mmcblk0p2 rootwait\0"                \
-		"bootcmd_mmc=run bootargs_base bootargs_mmc; bootm\0"   \
+		"bootcmd_mmc=run bootargs_base bootargs_mmc; "	\
+		"mmc dev 0; "	\
+		"mmc read ${loadaddr} 0x800 0x1800; bootm\0"   \
 		"bootcmd=run bootcmd_net\0"                             \
 
 
@@ -246,7 +248,6 @@
 	/* detect whether ESDHC1, ESDHC2, or ESDHC3 is boot device */
 	#define CONFIG_DYNAMIC_MMC_DEVNO
 
-	#define CONFIG_BOOT_PARTITION_ACCESS
 	#define CONFIG_EMMC_DDR_PORT_DETECT
 	#define CONFIG_EMMC_DDR_MODE
 
diff --git a/include/configs/mx50_arm2_ddr2.h b/include/configs/mx50_arm2_ddr2.h
index f2fa25c..cdf3b0e 100644
--- a/include/configs/mx50_arm2_ddr2.h
+++ b/include/configs/mx50_arm2_ddr2.h
@@ -237,7 +237,6 @@
 	/* detect whether ESDHC1, ESDHC2, or ESDHC3 is boot device */
 	#define CONFIG_DYNAMIC_MMC_DEVNO
 
-	#define CONFIG_BOOT_PARTITION_ACCESS
 	#define CONFIG_EMMC_DDR_PORT_DETECT
 	#define CONFIG_EMMC_DDR_MODE
 
diff --git a/include/configs/mx50_arm2_lpddr2.h b/include/configs/mx50_arm2_lpddr2.h
index 5fa3ef4..391ecf2 100644
--- a/include/configs/mx50_arm2_lpddr2.h
+++ b/include/configs/mx50_arm2_lpddr2.h
@@ -122,7 +122,9 @@
 			"tftpboot ${loadaddr} ${kernel}; bootm\0"	\
 		"bootargs_mmc=setenv bootargs ${bootargs} ip=dhcp "     \
 			"root=/dev/mmcblk0p2 rootwait\0"                \
-		"bootcmd_mmc=run bootargs_base bootargs_mmc; bootm\0"   \
+		"bootcmd_mmc=run bootargs_base bootargs_mmc; "	\
+		"mmc dev 0; "	\
+		"mmc read ${loadaddr} 0x800 0x1800; bootm\0"   \
 		"bootcmd=run bootcmd_net\0"                             \
 
 
@@ -238,7 +240,6 @@
 	/* detect whether ESDHC1, ESDHC2, or ESDHC3 is boot device */
 	#define CONFIG_DYNAMIC_MMC_DEVNO
 
-	#define CONFIG_BOOT_PARTITION_ACCESS
 	#define CONFIG_EMMC_DDR_PORT_DETECT
 	#define CONFIG_EMMC_DDR_MODE
 
diff --git a/include/configs/mx50_arm2_mfg.h b/include/configs/mx50_arm2_mfg.h
index a97ca0f..2dbddd0 100644
--- a/include/configs/mx50_arm2_mfg.h
+++ b/include/configs/mx50_arm2_mfg.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2010 Freescale Semiconductor, Inc.
+ * Copyright (C) 2010-2011 Freescale Semiconductor, Inc.
  *
  * Configuration settings for the MX50-ARM2 Freescale board.
  *
@@ -126,7 +126,9 @@
 			"tftpboot ${loadaddr} ${kernel}; bootm\0"	\
 		"bootargs_mmc=setenv bootargs ${bootargs} ip=dhcp "     \
 			"root=/dev/mmcblk0p2 rootwait\0"                \
-		"bootcmd_mmc=run bootargs_base bootargs_mmc; bootm\0"   \
+		"bootcmd_mmc=run bootargs_base bootargs_mmc; "   \
+		"mmc dev 0; "	\
+		"mmc read ${loadaddr} 0x800 0x1800; bootm\0"   \
 		"bootcmd=run bootcmd_net\0"                             \
 
 
@@ -206,9 +208,6 @@
 
 	/* detect whether ESDHC1 or ESDHC3 is boot device */
 	#define CONFIG_DYNAMIC_MMC_DEVNO
-
-	#define CONFIG_BOOT_PARTITION_ACCESS
-
 #endif
 /*-----------------------------------------------------------------------
  * Stack sizes
diff --git a/include/configs/mx50_rd3.h b/include/configs/mx50_rd3.h
index 904ddf7..51cc108 100644
--- a/include/configs/mx50_rd3.h
+++ b/include/configs/mx50_rd3.h
@@ -124,7 +124,9 @@
 			"tftpboot ${loadaddr} ${kernel}; bootm\0"	\
 		"bootargs_mmc=setenv bootargs ${bootargs} ip=dhcp "     \
 			"root=/dev/mmcblk0p2 rootwait\0"                \
-		"bootcmd_mmc=run bootargs_base bootargs_mmc; bootm\0"   \
+		"bootcmd_mmc=run bootargs_base bootargs_mmc; \0"   \
+		"mmc dev 0; "	\
+		"mmc read ${loadaddr} 0x800 0x1800; bootm\0"   \
 		"bootcmd=run bootcmd_net\0"                             \
 
 
@@ -215,7 +217,6 @@
 	/* detect whether ESDHC1, ESDHC2, or ESDHC3 is boot device */
 	#define CONFIG_DYNAMIC_MMC_DEVNO
 
-	#define CONFIG_BOOT_PARTITION_ACCESS
 	#define CONFIG_EMMC_DDR_PORT_DETECT
 	#define CONFIG_EMMC_DDR_MODE
 
diff --git a/include/configs/mx50_rd3_android.h b/include/configs/mx50_rd3_android.h
index 9082a0e..8c85040 100644
--- a/include/configs/mx50_rd3_android.h
+++ b/include/configs/mx50_rd3_android.h
@@ -120,8 +120,9 @@
 		"rd_loadaddr=0x70D00000\0"		\
 		"bootargs=console=ttymxc0 init=/init androidboot.console=ttymxc0 " \
 			"keypad video=mxc_elcdif_fb:off\0" \
-		"bootcmd_SD=mmc read 0 ${loadaddr} 0x800 0x2000;" \
-			"mmc read 0 ${rd_loadaddr} 0x3000 0x300\0" \
+		"bootcmd_SD=mmc dev 0;"	\
+			"mmc read ${loadaddr} 0x800 0x2000;" \
+			"mmc read ${rd_loadaddr} 0x3000 0x300\0" \
 		"bootcmd=run bootcmd_SD; bootm ${loadaddr} ${rd_loadaddr}\0" \
 
 
@@ -212,7 +213,6 @@
 	/* detect whether ESDHC1, ESDHC2, or ESDHC3 is boot device */
 	#define CONFIG_DYNAMIC_MMC_DEVNO
 
-	#define CONFIG_BOOT_PARTITION_ACCESS
 	#define CONFIG_EMMC_DDR_PORT_DETECT
 	#define CONFIG_EMMC_DDR_MODE
 
@@ -324,7 +324,8 @@
 	" rootfs=ext4 keypad"
 #define CONFIG_ANDROID_RECOVERY_BOOTCMD_MMC  \
 	"run bootargs_base bootargs_android_recovery;"  \
-	"mmc read 0 ${loadaddr} 0x800 0x2000;bootm"
+	"mmc dev 0"	\
+	"mmc read ${loadaddr} 0x800 0x2000;bootm"
 #define CONFIG_ANDROID_RECOVERY_CMD_FILE "/recovery/command"
 
 #define CONFIG_MTD_DEVICE
diff --git a/include/configs/mx50_rd3_mfg.h b/include/configs/mx50_rd3_mfg.h
index 9feda4a..55ab934 100644
--- a/include/configs/mx50_rd3_mfg.h
+++ b/include/configs/mx50_rd3_mfg.h
@@ -124,7 +124,9 @@
 			"tftpboot ${loadaddr} ${kernel}; bootm\0"	\
 		"bootargs_mmc=setenv bootargs ${bootargs} ip=dhcp "     \
 			"root=/dev/mmcblk0p2 rootwait\0"                \
-		"bootcmd_mmc=run bootargs_base bootargs_mmc; bootm\0"   \
+		"bootcmd_mmc=run bootargs_base bootargs_mmc; "   \
+		"mmc dev 0; "	\
+		"mmc read ${loadaddr} 0x800 0x1800; bootm\0"   \
 		"bootcmd=run bootcmd_net\0"                             \
 
 
@@ -193,9 +195,6 @@
 
 	/* detect whether ESDHC1 or ESDHC3 is boot device */
 	#define CONFIG_DYNAMIC_MMC_DEVNO
-
-	#define CONFIG_BOOT_PARTITION_ACCESS
-
 #endif
 /*-----------------------------------------------------------------------
  * Stack sizes
diff --git a/include/configs/mx50_rdp.h b/include/configs/mx50_rdp.h
index f89850c..ee381b0 100644
--- a/include/configs/mx50_rdp.h
+++ b/include/configs/mx50_rdp.h
@@ -124,7 +124,9 @@
 			"tftpboot ${loadaddr} ${kernel}; bootm\0"	\
 		"bootargs_mmc=setenv bootargs ${bootargs} ip=dhcp "     \
 			"root=/dev/mmcblk0p2 rootwait\0"                \
-		"bootcmd_mmc=run bootargs_base bootargs_mmc; bootm\0"   \
+		"bootcmd_mmc=run bootargs_base bootargs_mmc; "   \
+		"mmc dev 0; "	\
+		"mmc read ${loadaddr} 0x800 0x1800; bootm\0"   \
 		"bootcmd=run bootcmd_net\0"                             \
 
 
@@ -215,7 +217,6 @@
 	/* detect whether ESDHC1, ESDHC2, or ESDHC3 is boot device */
 	#define CONFIG_DYNAMIC_MMC_DEVNO
 
-	#define CONFIG_BOOT_PARTITION_ACCESS
 	#define CONFIG_EMMC_DDR_PORT_DETECT
 	#define CONFIG_EMMC_DDR_MODE
 
diff --git a/include/configs/mx50_rdp_android.h b/include/configs/mx50_rdp_android.h
index 10ed526..d234583 100644
--- a/include/configs/mx50_rdp_android.h
+++ b/include/configs/mx50_rdp_android.h
@@ -215,7 +215,6 @@
 	/* detect whether ESDHC1, ESDHC2, or ESDHC3 is boot device */
 	#define CONFIG_DYNAMIC_MMC_DEVNO
 
-	#define CONFIG_BOOT_PARTITION_ACCESS
 	#define CONFIG_EMMC_DDR_PORT_DETECT
 	#define CONFIG_EMMC_DDR_MODE
 
@@ -327,7 +326,8 @@
 	" rootfs=ext4 keypad"
 #define CONFIG_ANDROID_RECOVERY_BOOTCMD_MMC  \
 	"run bootargs_base bootargs_android_recovery;"  \
-	"mmc read 0 ${loadaddr} 0x800 0x2000;bootm"
+	"mmc dev 0"	\
+	"mmc read ${loadaddr} 0x800 0x2000;bootm"
 #define CONFIG_ANDROID_RECOVERY_CMD_FILE "/recovery/command"
 
 #define CONFIG_MTD_DEVICE
diff --git a/include/configs/mx50_rdp_mfg.h b/include/configs/mx50_rdp_mfg.h
index 7ca83b2..447cb35 100644
--- a/include/configs/mx50_rdp_mfg.h
+++ b/include/configs/mx50_rdp_mfg.h
@@ -124,7 +124,10 @@
 			"tftpboot ${loadaddr} ${kernel}; bootm\0"	\
 		"bootargs_mmc=setenv bootargs ${bootargs} ip=dhcp "     \
 			"root=/dev/mmcblk0p2 rootwait\0"                \
-		"bootcmd_mmc=run bootargs_base bootargs_mmc; bootm\0"   \
+		"bootcmd_mmc=run bootargs_base bootargs_mmc; "   \
+		"mmc dev 0; "	\
+		"mmc read ${loadaddr} 0x800 0x1800; bootm\0"   \
+
 		"bootcmd=run bootcmd_net\0"                             \
 
 
@@ -193,9 +196,6 @@
 
 	/* detect whether ESDHC1 or ESDHC3 is boot device */
 	#define CONFIG_DYNAMIC_MMC_DEVNO
-
-	#define CONFIG_BOOT_PARTITION_ACCESS
-
 #endif
 /*-----------------------------------------------------------------------
  * Stack sizes
diff --git a/include/configs/mx51_3stack_android.h b/include/configs/mx51_3stack_android.h
index 8e0909d..85b6251 100644
--- a/include/configs/mx51_3stack_android.h
+++ b/include/configs/mx51_3stack_android.h
@@ -131,7 +131,7 @@
 #define CONFIG_ANDROID_RECOVERY_BOOTARGS_MMC \
 	"setenv bootargs ${bootargs} root=/dev/mmcblk0p4 ip=off init=/init rootfstype=ext3"
 #define CONFIG_ANDROID_RECOVERY_BOOTCMD_MMC  \
-	"run bootargs_base bootargs_android;mmc read 0 ${loadaddr} 0x800 0x1280;bootm"
+	"run bootargs_base bootargs_android;mmc dev 0;mmc read ${loadaddr} 0x800 0x1280;bootm"
 #define CONFIG_ANDROID_RECOVERY_BOOTARGS_NAND \
 	"setenv bootargs ${bootargs} ip=off rootfstype=ubifs root=ubi1:recovery init=/init ubi.mtd=3 ubi.mtd=2"
 #define CONFIG_ANDROID_RECOVERY_BOOTCMD_NAND  \
@@ -186,8 +186,9 @@
 		"bootcmd_net=run bootargs_base bootargs_nfs; "		\
 			"tftpboot ${loadaddr} ${kernel}; bootm\0"	\
 		"bootcmd_android=run bootargs_base bootargs_android; "	\
-			"mmc read 0 ${loadaddr} 0x800 0x1280; "	\
-			"mmc read 0 ${rd_loadaddr} 0x2000 0x258; "	\
+			"mmc dev 0"					\
+			"mmc read ${loadaddr} 0x800 0x1280; "		\
+			"mmc read ${rd_loadaddr} 0x2000 0x258; "	\
 			"bootm ${loadaddr} ${rd_loadaddr}\0"		\
 		"prg_uboot=tftpboot ${loadaddr} ${uboot}; "		\
 			"protect off ${uboot_addr} 0xa003ffff; "	\
diff --git a/include/configs/mx51_bbg_android.h b/include/configs/mx51_bbg_android.h
index cfa8650..df99cbc 100644
--- a/include/configs/mx51_bbg_android.h
+++ b/include/configs/mx51_bbg_android.h
@@ -126,7 +126,8 @@
 	"rootfs=ext4 di1_primary"
 #define CONFIG_ANDROID_RECOVERY_BOOTCMD_MMC  \
 	"run bootargs_base bootargs_android_recovery;"	\
-	"mmc read 0 ${loadaddr} 0x800 0x2000;bootm"
+	"mmc dev 0"	\
+	"mmc read ${loadaddr} 0x800 0x2000;bootm"
 #define CONFIG_ANDROID_RECOVERY_CMD_FILE "/recovery/command"
 
 #define CONFIG_ANDROID_SYSTEM_PARTITION_MMC 2
@@ -170,8 +171,9 @@
 		"rd_loadaddr=0x90D00000\0"		\
 		"bootargs=console=ttymxc0 init=/init androidboot.console=ttymxc0 " \
 			"di1_primary gpu_nommu gpu_memory=32M calibration\0" \
-		"bootcmd_SD=mmc read 0 ${loadaddr} 0x800 0x2000;" \
-			"mmc read 0 ${rd_loadaddr} 0x3000 0x300\0" \
+		"bootcmd_SD=mmc dev 0;"	\
+			"mmc read ${loadaddr} 0x800 0x2000;" \
+			"mmc read ${rd_loadaddr} 0x3000 0x300\0" \
 		"bootcmd=run bootcmd_SD; bootm ${loadaddr} ${rd_loadaddr}\0" \
 
 
diff --git a/include/configs/mx53_ard.h b/include/configs/mx53_ard.h
index 74661c3..c7d8ddb 100644
--- a/include/configs/mx53_ard.h
+++ b/include/configs/mx53_ard.h
@@ -135,7 +135,8 @@
 		"bootargs_mmc=setenv bootargs ${bootargs} ip=dhcp "     \
 			"root=/dev/mmcblk0p1 rootwait rw\0"                \
 		"bootcmd_mmc=run bootargs_base bootargs_mmc; "		\
-			"mmc read 0 ${loadaddr} 0x800 0x1800; bootm\0"   \
+			"mmc dev 0"					\
+			"mmc read ${loadaddr} 0x800 0x1800; bootm\0"   \
 		"bootcmd=run bootcmd_net\0"                             \
 
 
@@ -222,8 +223,6 @@
 
 	/* detect whether ESDHC1 or ESDHC3 is boot device */
 	#define CONFIG_DYNAMIC_MMC_DEVNO
-
-	#define CONFIG_BOOT_PARTITION_ACCESS
 #endif
 
 /*
diff --git a/include/configs/mx53_ard_ddr3.h b/include/configs/mx53_ard_ddr3.h
index a3f348f..569c7d1 100644
--- a/include/configs/mx53_ard_ddr3.h
+++ b/include/configs/mx53_ard_ddr3.h
@@ -135,7 +135,8 @@
 		"bootargs_mmc=setenv bootargs ${bootargs} ip=dhcp "     \
 			"root=/dev/mmcblk0p1 rootwait rw\0"                \
 		"bootcmd_mmc=run bootargs_base bootargs_mmc; "		\
-			"mmc read 0 ${loadaddr} 0x800 0x1800; bootm\0"   \
+			"mmc dev 0"	\
+			"mmc read ${loadaddr} 0x800 0x1800; bootm\0"   \
 		"bootcmd=run bootcmd_net\0"                             \
 
 
@@ -222,8 +223,6 @@
 
 	/* detect whether ESDHC1 or ESDHC3 is boot device */
 	#define CONFIG_DYNAMIC_MMC_DEVNO
-
-	#define CONFIG_BOOT_PARTITION_ACCESS
 #endif
 
 /*
diff --git a/include/configs/mx53_arm2_ddr3.h b/include/configs/mx53_arm2_ddr3.h
index a21f5ea..6074563 100644
--- a/include/configs/mx53_arm2_ddr3.h
+++ b/include/configs/mx53_arm2_ddr3.h
@@ -125,7 +125,9 @@
 			"tftpboot ${loadaddr} ${kernel}; bootm\0"	\
 		"bootargs_mmc=setenv bootargs ${bootargs} ip=dhcp "     \
 			"root=/dev/mmcblk0p2 rootwait\0"                \
-		"bootcmd_mmc=run bootargs_base bootargs_mmc; bootm\0"   \
+		"bootcmd_mmc=run bootargs_base bootargs_mmc; "   \
+		"mmc dev 0; "	\
+		"mmc read ${loadaddr} 0x800 0x1800; bootm\0"	\
 		"bootcmd=run bootcmd_net\0"                             \
 
 
@@ -213,7 +215,6 @@
 	#define CONFIG_DOS_PARTITION	1
 	#define CONFIG_CMD_FAT		1
 	#define CONFIG_CMD_EXT2		1
-	#define CONFIG_BOOT_PARTITION_ACCESS
 	#define CONFIG_EMMC_DDR_PORT_DETECT
 	#define CONFIG_DYNAMIC_MMC_DEVNO
 	#define CONFIG_EMMC_DDR_MODE
diff --git a/include/configs/mx53_arm2_ddr3_android.h b/include/configs/mx53_arm2_ddr3_android.h
index 30edc70..b88f227 100644
--- a/include/configs/mx53_arm2_ddr3_android.h
+++ b/include/configs/mx53_arm2_ddr3_android.h
@@ -86,7 +86,7 @@
 #define CONFIG_ANDROID_RECOVERY_BOOTARGS_MMC \
 	"setenv bootargs ${bootargs} init=/init root=/dev/mmcblk0p4 rootfs=ext4"
 #define CONFIG_ANDROID_RECOVERY_BOOTCMD_MMC  \
-	"run bootargs_base bootargs_android_recovery;mmc read 0 ${loadaddr} 0x800 0x1800;bootm"
+	"run bootargs_base bootargs_android_recovery;mmc dev 0;mmc read ${loadaddr} 0x800 0x1800;bootm"
 #define CONFIG_ANDROID_RECOVERY_CMD_FILE "/recovery/command"
 #define CONFIG_ANDROID_CACHE_PARTITION_MMC 6
 
@@ -140,22 +140,24 @@
 		"bootargs_base=setenv bootargs ${bootargs} "		\
 		"console=ttymxc0,115200\0"				\
 		"bootargs_nfs=setenv bootargs ${bootargs} root=/dev/nfs " \
-		     "ip=dhcp nfsroot=${serverip}:${nfsroot},v3,tcp\0"	\
+			"ip=dhcp nfsroot=${serverip}:${nfsroot},v3,tcp\0"	\
 		"bootargs_android=setenv bootargs ${bootargs} mem=512M " \
-		     "androidboot.console=ttymxc0 init=/init "		\
-		     "video=mxcdi0fb:RGB565,800x480M@55 calibration\0"	\
+			"androidboot.console=ttymxc0 init=/init "		\
+			"video=mxcdi0fb:RGB565,800x480M@55 calibration\0"	\
 		"bootcmd=run bootcmd_SD \0"				\
 		"bootcmd_SD=run bootargs_base bootargs_android;"	\
-		     "mmc read 0 ${loadaddr} 0x800 1800;"		\
-		     "mmc read 0 ${rd_loadaddr} 0x2000 0x258;"		\
-		     "bootm ${loadaddr} ${rd_loadaddr}\0"		\
+			"mmc dev 0"					\
+			"mmc read ${loadaddr} 0x800 1800;"		\
+			"mmc read ${rd_loadaddr} 0x2000 0x258;"		\
+			"bootm ${loadaddr} ${rd_loadaddr}\0"		\
 		"bootcmd_net=run bootargs_base bootargs_nfs; "		\
 			"tftpboot ${loadaddr} ${kernel}; bootm\0"	\
 		"bootcmd_android_recovery=run bootargs_base"		\
-		     " bootargs_android_recovery;"			\
-		     "mmc read 0 ${loadaddr} 0x800 0x1800;bootm\0"	\
+			" bootargs_android_recovery;"			\
+			"mmc dev 0"					\
+			"mmc read ${loadaddr} 0x800 0x1800;bootm\0"	\
 		"bootargs_android_recovery=setenv bootargs ${bootargs}" \
-		     " init=/init root=/dev/mmcblk0p4 rootfs=ext4\0"	\
+			" init=/init root=/dev/mmcblk0p4 rootfs=ext4\0"	\
 
 #define CONFIG_ARP_TIMEOUT	200UL
 
@@ -241,7 +243,6 @@
 	#define CONFIG_DOS_PARTITION	1
 	#define CONFIG_CMD_FAT		1
 	#define CONFIG_CMD_EXT2		1
-	#define CONFIG_BOOT_PARTITION_ACCESS
 	#define CONFIG_EMMC_DDR_PORT_DETECT
 	#define CONFIG_EMMC_DDR_MODE
 	/* port 1 (ESDHC3) is 8 bit */
diff --git a/include/configs/mx53_evk.h b/include/configs/mx53_evk.h
index 5db9643..da4ff4e 100644
--- a/include/configs/mx53_evk.h
+++ b/include/configs/mx53_evk.h
@@ -134,7 +134,10 @@
 			"tftpboot ${loadaddr} ${kernel}; bootm\0"	\
 		"bootargs_mmc=setenv bootargs ${bootargs} ip=dhcp "     \
 			"root=/dev/mmcblk0p2 rootwait\0"                \
-		"bootcmd_mmc=run bootargs_base bootargs_mmc; bootm\0"   \
+		"bootcmd_mmc=run bootargs_base bootargs_mmc; "   \
+		"mmc dev 0; "	\
+		"mmc read ${loadaddr} 0x800 0x1800; bootm\0"	\
+
 		"bootcmd=run bootcmd_net\0"                             \
 
 
@@ -226,7 +229,6 @@
 	/* detect whether ESDHC1 or ESDHC3 is boot device */
 	#define CONFIG_DYNAMIC_MMC_DEVNO
 
-	#define CONFIG_BOOT_PARTITION_ACCESS
 	#define CONFIG_EMMC_DDR_PORT_DETECT
 	#define CONFIG_EMMC_DDR_MODE
 	/* port 1 (ESDHC3) is 8 bit */
diff --git a/include/configs/mx53_evk_android.h b/include/configs/mx53_evk_android.h
index dbad007..4f37e4d 100644
--- a/include/configs/mx53_evk_android.h
+++ b/include/configs/mx53_evk_android.h
@@ -102,7 +102,7 @@
 #define CONFIG_ANDROID_RECOVERY_BOOTARGS_MMC \
 	"setenv bootargs ${bootargs} init=/init root=/dev/mmcblk0p4 rootfs=ext4"
 #define CONFIG_ANDROID_RECOVERY_BOOTCMD_MMC  \
-	"run bootargs_base bootargs_android_recovery;mmc read 0 ${loadaddr} 0x800 0x2000;bootm"
+	"run bootargs_base bootargs_android_recovery;mmc dev 0;mmc read ${loadaddr} 0x800 0x2000;bootm"
 #define CONFIG_ANDROID_RECOVERY_CMD_FILE "/recovery/command"
 
 #define CONFIG_ANDROID_SYSTEM_PARTITION_MMC 2
@@ -169,22 +169,24 @@
 		"bootargs_base=setenv bootargs ${bootargs} "		\
 		"console=ttymxc0,115200 gpu_memory=16M\0"		\
 		"bootargs_nfs=setenv bootargs ${bootargs} root=/dev/nfs " \
-		     "ip=dhcp nfsroot=${serverip}:${nfsroot},v3,tcp\0"	\
+			"ip=dhcp nfsroot=${serverip}:${nfsroot},v3,tcp\0"	\
 		"bootargs_android=setenv bootargs ${bootargs} " \
-		     "androidboot.console=ttymxc0 init=/init di0_primary" \
-		     "video=mxcdi0fb:RGB565,800x480M@55 calibration\0"	\
+			"androidboot.console=ttymxc0 init=/init di0_primary" \
+			"video=mxcdi0fb:RGB565,800x480M@55 calibration\0"	\
 		"bootcmd=run bootcmd_SD \0"				\
 		"bootcmd_SD=run bootargs_base bootargs_android;"	\
-		     "mmc read 0 ${loadaddr} 0x800 2000;"		\
-		     "mmc read 0 ${rd_loadaddr} 0x3000 0x258;"		\
-		     "bootm ${loadaddr} ${rd_loadaddr}\0"		\
+			"mmc dev 0"					\
+			"mmc read ${loadaddr} 0x800 2000;"		\
+			"mmc read ${rd_loadaddr} 0x3000 0x258;"		\
+			"bootm ${loadaddr} ${rd_loadaddr}\0"		\
 		"bootcmd_net=run bootargs_base bootargs_nfs; "		\
 			"tftpboot ${loadaddr} ${kernel}; bootm\0"	\
 		"bootcmd_android_recovery=run bootargs_base"		\
-		     " bootargs_android_recovery;"			\
-		     "mmc read 0 ${loadaddr} 0x800 0x2000;bootm\0"	\
+			" bootargs_android_recovery;"			\
+			"mmc dev 0"					\
+			"mmc read ${loadaddr} 0x800 0x2000;bootm\0"	\
 		"bootargs_android_recovery=setenv bootargs ${bootargs}" \
-		     " init=/init root=/dev/mmcblk0p4 rootfs=ext4\0"	\
+			" init=/init root=/dev/mmcblk0p4 rootfs=ext4\0"	\
 
 #define CONFIG_ARP_TIMEOUT	200UL
 
@@ -274,8 +276,6 @@
 	/* detect whether ESDHC1 or ESDHC3 is boot device */
 	#define CONFIG_DYNAMIC_MMC_DEVNO
 
-
-	#define CONFIG_BOOT_PARTITION_ACCESS
 	#define CONFIG_EMMC_DDR_PORT_DETECT
 	#define CONFIG_EMMC_DDR_MODE
 	/* port 1 (ESDHC3) is 8 bit */
diff --git a/include/configs/mx53_evk_mfg.h b/include/configs/mx53_evk_mfg.h
index 33f0a5e..e4c666e 100755
--- a/include/configs/mx53_evk_mfg.h
+++ b/include/configs/mx53_evk_mfg.h
@@ -188,7 +188,6 @@
 	#define CONFIG_CMD_FAT		1
 	#define CONFIG_CMD_EXT2		1
 
-	#define CONFIG_BOOT_PARTITION_ACCESS
 	#define CONFIG_EMMC_DDR_PORT_DETECT
 	#define CONFIG_EMMC_DDR_MODE
 	/* port 1 (ESDHC3) is 8 bit */
diff --git a/include/configs/mx53_loco.h b/include/configs/mx53_loco.h
index e9cb29b..75ea8e9 100644
--- a/include/configs/mx53_loco.h
+++ b/include/configs/mx53_loco.h
@@ -127,7 +127,8 @@
 		"bootargs_mmc=setenv bootargs ${bootargs} ip=dhcp "	\
 			"root=/dev/mmcblk0p1 rootwait rw\0"		\
 		"bootcmd_mmc=run bootargs_base bootargs_mmc; "		\
-			"mmc read 0 ${loadaddr} 0x800 0x1800; bootm\0"	\
+			"mmc dev 0"					\
+			"mmc read ${loadaddr} 0x800 0x1800; bootm\0"	\
 		"bootcmd=run bootcmd_mmc\0"	\
 	\
 
@@ -207,7 +208,6 @@
 	/* detect whether ESDHC1 or ESDHC3 is boot device */
 	#define CONFIG_DYNAMIC_MMC_DEVNO
 
-	#define CONFIG_BOOT_PARTITION_ACCESS
 	#define CONFIG_EMMC_DDR_PORT_DETECT
 	#define CONFIG_EMMC_DDR_MODE
 	/* port 1 (ESDHC3) is 8 bit */
diff --git a/include/configs/mx53_loco_mfg.h b/include/configs/mx53_loco_mfg.h
index b6ceb65..bdaa081 100644
--- a/include/configs/mx53_loco_mfg.h
+++ b/include/configs/mx53_loco_mfg.h
@@ -193,7 +193,6 @@
 	/* detect whether ESDHC1 or ESDHC3 is boot device */
 	#define CONFIG_DYNAMIC_MMC_DEVNO
 
-	#define CONFIG_BOOT_PARTITION_ACCESS
 	#define CONFIG_EMMC_DDR_PORT_DETECT
 	#define CONFIG_EMMC_DDR_MODE
 	/* port 1 (ESDHC3) is 8 bit */
diff --git a/include/configs/mx53_pcba_android.h b/include/configs/mx53_pcba_android.h
index 65b58f8..3164b38 100644
--- a/include/configs/mx53_pcba_android.h
+++ b/include/configs/mx53_pcba_android.h
@@ -104,7 +104,7 @@
 	"setenv bootargs ${bootargs} init=/init root=/dev/mmcblk0p4 rootfs=ext4 video=mxcdi1fb:RGB666,XGA ldb=di1 di1_primary"
 #define CONFIG_ANDROID_RECOVERY_BOOTCMD_MMC  \
 	"run bootargs_base bootargs_android_recovery;"  \
-	"mmc read 0 ${loadaddr} 0x800 0x2000;bootm"
+	"mmc dev 0; read ${loadaddr} 0x800 0x2000;bootm"
 #define CONFIG_ANDROID_RECOVERY_CMD_FILE "/recovery/command"
 
 #define CONFIG_ANDROID_SYSTEM_PARTITION_MMC 2
@@ -146,8 +146,8 @@
 		"bootargs=console=ttymxc0 init=/init " \
 			"androidboot.console=ttymxc0 video=mxcdi0fb:RGB24,AT070TN93 " \
 			"di0_primary gpu_nommu gpu_memory=32M\0" \
-		"bootcmd_SD=mmc read 1 ${loadaddr} 0x800 0x2000;" \
-			"mmc read 1 ${rd_loadaddr} 0x3000 0x300\0" \
+		"bootcmd_SD=mmc dev 1 0; mmc read ${loadaddr} 0x800 0x2000;" \
+			"mmc read ${rd_loadaddr} 0x3000 0x300\0" \
 		"bootcmd=run bootcmd_SD; bootm ${loadaddr} ${rd_loadaddr}\0" \
 
 
diff --git a/include/configs/mx53_smd.h b/include/configs/mx53_smd.h
index f19ea50..2ed27ab 100644
--- a/include/configs/mx53_smd.h
+++ b/include/configs/mx53_smd.h
@@ -130,7 +130,8 @@
 		"bootargs_mmc=setenv bootargs ${bootargs} ip=dhcp "	\
 			"root=/dev/mmcblk0p1 rootwait rw\0"		\
 		"bootcmd_mmc=run bootargs_base bootargs_mmc; "		\
-			"mmc read 0 ${loadaddr} 0x800 0x1800; bootm\0"	\
+		"mmc dev 0; "	\
+		"mmc read ${loadaddr} 0x800 0x1800; bootm\0"	\
 		"bootcmd=run bootcmd_mmc\0"	\
 	\
 
diff --git a/include/configs/mx53_smd_android.h b/include/configs/mx53_smd_android.h
index b9d6d47..2830932 100644
--- a/include/configs/mx53_smd_android.h
+++ b/include/configs/mx53_smd_android.h
@@ -104,7 +104,8 @@
 	"setenv bootargs ${bootargs} init=/init root=/dev/mmcblk0p4 rootfs=ext4 video=mxcdi1fb:RGB666,XGA ldb=di1 di1_primary"
 #define CONFIG_ANDROID_RECOVERY_BOOTCMD_MMC  \
 	"run bootargs_base bootargs_android_recovery;"	\
-	"mmc read 1 ${loadaddr} 0x800 0x2000;bootm"
+	"mmc dev 1 0"	\
+	"mmc read ${loadaddr} 0x800 0x2000;bootm"
 #define CONFIG_ANDROID_RECOVERY_CMD_FILE "/recovery/command"
 
 #define CONFIG_ANDROID_SYSTEM_PARTITION_MMC 2
@@ -164,8 +165,9 @@
 		"bootargs=console=ttymxc0 init=/init " \
 			"androidboot.console=ttymxc0 video=mxcdi1fb:RGB666,XGA " \
 			"ldb=di1 di1_primary gpu_nommu gpu_memory=64M\0" \
-		"bootcmd_SD=mmc read 1 ${loadaddr} 0x800 0x2000;" \
-			"mmc read 1 ${rd_loadaddr} 0x3000 0x300\0" \
+		"bootcmd_SD=mmc dev 1 0;"		\
+			"mmc read ${loadaddr} 0x800 0x2000;" \
+			"mmc read ${rd_loadaddr} 0x3000 0x300\0" \
 		"bootcmd=run bootcmd_SD; bootm ${loadaddr} ${rd_loadaddr}\0" \
 
 #define CONFIG_ARP_TIMEOUT	200UL
@@ -257,7 +259,6 @@
 	/* detect whether ESDHC1 or ESDHC3 is boot device */
 	#define CONFIG_DYNAMIC_MMC_DEVNO
 
-	#define CONFIG_BOOT_PARTITION_ACCESS
 	#define CONFIG_EMMC_DDR_MODE
 	#define CONFIG_EMMC_DDR_PORT_DETECT
 	/* port 1 (ESDHC3) is 8 bit */
diff --git a/include/configs/mx53_smd_mfg.h b/include/configs/mx53_smd_mfg.h
index 8fe8811..2d1e2e0 100644
--- a/include/configs/mx53_smd_mfg.h
+++ b/include/configs/mx53_smd_mfg.h
@@ -210,7 +210,6 @@
 	/* detect whether ESDHC1 or ESDHC3 is boot device */
 	#define CONFIG_DYNAMIC_MMC_DEVNO
 
-	#define CONFIG_BOOT_PARTITION_ACCESS
 	#define CONFIG_EMMC_DDR_MODE
 	#define CONFIG_EMMC_DDR_PORT_DETECT
 	/* port 1 (ESDHC3) is 8 bit */
diff --git a/include/configs/mx6q_sabreauto.h b/include/configs/mx6q_sabreauto.h
index 32247c0..c310155 100644
--- a/include/configs/mx6q_sabreauto.h
+++ b/include/configs/mx6q_sabreauto.h
@@ -119,7 +119,9 @@
 			"tftpboot ${loadaddr} ${kernel}; bootm\0"	\
 		"bootargs_mmc=setenv bootargs ${bootargs} ip=dhcp "     \
 			"root=/dev/mmcblk0p1 rootwait\0"                \
-		"bootcmd_mmc=run bootargs_base bootargs_mmc; bootm\0"   \
+		"bootcmd_mmc=run bootargs_base bootargs_mmc; "   \
+		"mmc dev 0; "	\
+		"mmc read ${loadaddr} 0x800 0x1800; bootm\0"	\
 		"bootcmd=run bootcmd_net\0"                             \
 
 
@@ -180,7 +182,6 @@
 	/* detect whether SD1, 2, 3, or 4 is boot device */
 	#define CONFIG_DYNAMIC_MMC_DEVNO
 
-	#define CONFIG_BOOT_PARTITION_ACCESS
 	/* SD3 and SD4 are 8 bit */
 	#define CONFIG_MMC_8BIT_PORTS   0xC
 	/* Setup target delay in DDR mode for each SD port */
-- 
1.7.7.4

